### PR TITLE
backup: add sort unit test

### PR DESCRIPTION
**Pull request by xiang90**
/cc @hongchaodeng

### CODE REVIEW

**Comment by hongchaodeng** [comment]
@etcd-bot retest this please

File: pkg/backup/util_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+)
+
+func TestFilterAndSortBackups(t *testing.T) {
+	names := []string{
+		"3.0.1_12_etcd.backup",
+		"3.0.1_18_etcd.tmp",
+		"3.0.1_3_etcd.backup",
+		"3.0.3_19_etcd.backup",
+		"3.0.0_1_etcd.backup",
+		"3.0.1_badbackup_etcd.backup"}
```
Comment (hongchaodeng) [inline pkg/backup/util_test.go]: put "}" in next line?

File: pkg/backup/util_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestFilterAndSortBackups(t *testing.T) {
+	names := []string{
+		"3.0.1_12_etcd.backup",
+		"3.0.1_18_etcd.tmp",
```
Comment (hongchaodeng) [inline pkg/backup/util_test.go]: put a comment afterwards to note it:
```
"3.0.1_18_etcd.tmp", // wrong suffix
```

File: pkg/backup/util_test.go (hunk 1 above)
Comment (hongchaodeng) [inline pkg/backup/util_test.go]: Put a comment afterwards to note it:
```
"3.0.1_badbackup_etcd.backup", // malformed name
```

**Comment by xiang90** [comment]
All fixed.

**Comment by hongchaodeng** [comment]
lgtm

**Comment by hongchaodeng** [comment]
Pull it down and run unit test on linux machine manually.
Since it passes, we can just merge it.

**Pull request merged**